### PR TITLE
Rapidfire improvements

### PR DESCRIPTION
**Pull request by BenCat07**
- Add Instant zoom (and Heatmaker support)
- Fix misses while moving and using Rapidfire
- Improve Performance while using Rapidfire, alot

might want to squish these commits

### CODE REVIEW

File: src/core/interfaces.cpp [hunk 1]
```diff
@@ -169,6 +170,7 @@ void CreateInterfaces()
     g_IStudioRender     = BruteforceInterface<IStudioRender>("VStudioRender", sharedobj::studiorender());
     g_IVRenderView      = BruteforceInterface<IVRenderView>("VEngineRenderView", sharedobj::engine());
     g_IMaterialSystemHL = (IMaterialSystem *) g_IMaterialSystem;
+    g_IToolFramework    = BruteforceInterface<IToolFrameworkInternal>("VTOOLFRAMEWORKVERSION", sharedobj::engine());
```
Comment (BenCat07) [inline src/core/interfaces.cpp:173]: hey idiot, this is not visual only

**Comment by delimeats-ch** [comment]
Very nice.

File: src/core/interfaces.cpp (hunk 1 above)
Comment (TotallyNotElite) [reply src/core/interfaces.cpp:173]: Uh

**Review by TotallyNotElite — DISMISSED** [review]
IMO this entire approach is gay

File: src/core/interfaces.cpp (hunk 1 above)
Comment (BenCat07) [reply src/core/interfaces.cpp:173]: i need to fix this for textmode

**Pull request merged**